File: exts/jphp-gui-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/UXGamePane.java
package org.develnext.jphp.ext.game.classes;

import javafx.scene.Node;
import javafx.scene.layout.AnchorPane;
import javafx.scene.paint.Color;
import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.GamePane;
import org.develnext.jphp.ext.javafx.classes.layout.UXAnchorPane;
import org.develnext.jphp.ext.javafx.classes.layout.UXPanel;
import org.develnext.jphp.ext.javafx.classes.layout.UXScrollPane;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Nullable;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Namespace(GameExtension.NS)
public class UXGamePane extends UXPanel<GamePane> {
    interface WrappedInterface {
        @Property Node content();

        @Property double viewX();
        @Property double viewY();
        @Property double viewWidth();
        @Property double viewHeight();

        void scrollTo(double x, double y);
        void loadArea(AnchorPane area);
    }

    public UXGamePane(Environment env, GamePane wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGamePane(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public void __construct() {
        __wrappedObject = new GamePane();
    }

    @Override
    protected double getWidth() {
        return getWrappedObject().getViewWidth();
    }

    @Override
    protected double getHeight() {
        return getWrappedObject().getViewHeight();
    }

    @Override
    protected double[] getSize() {
        return new double[] { getWidth(), getHeight() };
    }

    @Override
    public GamePane getWrappedObject() {
        return (GamePane) super.getWrappedObject();
    }
}

File: exts/jphp-gui-game-ext/src/main/resources/JPHP-INF/sdk/php/game/UXGamePane.php
<?php
namespace php\game;

use php\gui\layout\UXAnchorPane;
use php\gui\layout\UXPanel;
use php\gui\layout\UXScrollPane;
use php\gui\paint\UXColor;
use php\gui\UXNode;

/**
 * Class UXGamePane
 * @package php\game
 * @packages game, javafx
 */
class UXGamePane extends UXPanel
{
    /**
     * @var UXGameScene
     */
    public $scene;

    /**
     * @var UXAnchorPane
     */
    public $content;

    /**
     * @var bool
     */
    public $autoSize = false;

    /**
     * @var string|UXColor
     */
    public $areaBackgroundColor = 'white';

    /**
     * @readonly
     * @var int
     */
    public $viewX = 0, $viewY = 0;

    /**
     * @var float
     */
    public $viewWidth = 0.0, $viewHeight = 0.0;

    /**
     * @param $x
     * @param $y
     */
    public function scrollTo($x, $y)
    {
    }

    /**
     * @param UXAnchorPane $area
     */
    public function loadArea(UXAnchorPane $area)
    {
    }
}